### DISCUSSION TITLE
How to prevent from setting pageIndex to 0?

### DISCUSSION THREAD

**Discussion by qwexvf** [KevinVandy/material-react-table#440] [General]
```ts
  const [pagination, setPagination] = useState<MRT_PaginationState>({
    pageIndex: 1,
    pageSize: 10
  })
...
  <MaterialReactTable
    ....
    state={{
       pagination
    }}
    onPaginationChange={setPagination}
  />
```

this works but i need my `pageIndex` always to be `1` how can i achieve this?

**Comment by qwexvf** [comment]
https://github.com/KevinVandy/material-react-table/blob/32ae8f925112a09298d3f4a5b3a0649701b2c2d0/apps/material-react-table-docs/examples/external-toolbar/sandbox/src/TS.tsx#L144

Going to test this

**Reply by KevinVandy** [reply]
autoResetPageIndex=false

**Reply by samchiu90** [reply]
Hi @KevinVandy , I have updated the package from 1.9.2 to 1.9.3 (also 1.11.3) and I find that onPaginationChange still reset the Pagination after starting fetching data even though I have set manualPagination &  autoResetPageIndex={false}
This issue does not occur before being updated to >= 1.9.3.
```
<MaterialReactTable
  /**/
  state={{
    ...ColumnsConfig,
    pagination,
    sorting: sorts,
    columnFilters,
    columnFilterFns,
    isLoading: isLoadingData,
  }}
  enablePagination
  manualPagination
  autoResetPageIndex={false}
  rowCount={total}
  muiTablePaginationProps={{
    rowsPerPageOptions: [10, 15, 20, 25, 50, 100],
    count: total,
  }}
  onPaginationChange={(updater) => {
    setPagination((prev) => {
          console.debug({
            updater,
            prev,
            new: updater instanceof Function ? updater(prev) : updater,
          });// <-- checking
      return updater instanceof Function ? updater(prev) : updater;
    });
  }}
  positionPagination="both"
/>    
```
After I change the state of the Pagination's pageIndex from 0 to 1, the behavior is normal and it will fetch data.
After start fetching data, I see onPaginationChange reset the Pagination from
```
prev: {
    "pageIndex": 1,
    "pageSize": 20
}

``` 
to
```
new: {
    "pageIndex": 0,
    "pageSize": 20
}
```

**Reply by samchiu90** [reply]
Ok, I solve my problem
I need to set the pagination in the state like this:
 ```
       state={{
        /*...*/
        pagination: {
          ...pagination,
          pageIndex: !total || total <= 0 ? 0 : pagination.pageIndex,
        },
        isLoading: isLoadingData,
      }}
 ```
`total` is the total number of rows
https://stackoverflow.com/questions/61840990/failed-prop-type-material-ui-the-page-prop-of-a-tablepagination-is-out-of-rang
